looks like the new image wasn't pulled last time round. Fixing that I reran the installation and encountered the same issue:
microbedb repository is installed, but not the tool

```
galaxy.tool_shed.galaxy_install.repository_dependencies.repository_dependency_manager DEBUG 2020-11-09 01:08:13,803 [p:34,w:1,m:0] [uWSGIWorker1Core2] Building repository dependency relationships...
galaxy.tool_shed.util.repository_util DEBUG 2020-11-09 01:08:13,851 [p:34,w:1,m:0] [uWSGIWorker1Core1] Updating an existing row for repository 'microbedb' in the tool_shed_repository table, status set to 'New'.
galaxy.tool_shed.galaxy_install.repository_dependencies.repository_dependency_manager DEBUG 2020-11-09 01:08:13,978 [p:34,w:1,m:0] [uWSGIWorker1Core1] Building repository dependency relationships...
galaxy.tool_shed.galaxy_install.install_manager INFO 2020-11-09 01:08:14,198 [p:34,w:1,m:0] [uWSGIWorker1Core2] Cloning repository 'https://toolshed.g2.bx.psu.edu/repos/brinkmanlab/awkscript' at 4:7966a43dbc9e
galaxy.tool_shed.util.repository_util DEBUG 2020-11-09 01:08:14,356 [p:34,w:1,m:0] [uWSGIWorker1Core0] Updating an existing row for repository 'ncbi_blast_plus' in the tool_shed_repository table, status set to 'New'.
galaxy.tool_shed.galaxy_install.install_manager INFO 2020-11-09 01:08:14,447 [p:34,w:1,m:0] [uWSGIWorker1Core1] Cloning repository 'https://toolshed.g2.bx.psu.edu/repos/brinkmanlab/microbedb' at 4:69186a3c9e76
galaxy.tool_shed.galaxy_install.repository_dependencies.repository_dependency_manager DEBUG 2020-11-09 01:08:14,528 [p:34,w:1,m:0] [uWSGIWorker1Core0] Building repository dependency relationships...
galaxy.tool_shed.galaxy_install.install_manager INFO 2020-11-09 01:08:15,830 [p:34,w:1,m:0] [uWSGIWorker1Core0] Cloning repository 'https://toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus' at 26:2889433c7ae1
galaxy.util ERROR 2020-11-09 01:08:25,704 [p:34,w:1,m:0] [uWSGIWorker1Core1] ERROR: Unable to read builds file: [Errno 2] No such file or directory: 'tool-data/shared/ucsc/builds.txt.sample'
galaxy.tools.data WARNING 2020-11-09 01:08:25,779 [p:34,w:1,m:0] [uWSGIWorker1Core1] Cannot find index file '/data/database/tmp/tmp-toolshed-gmfcr71060v2u/all_fasta.loc' for tool data table 'all_fasta'
galaxy.tools.data DEBUG 2020-11-09 01:08:25,780 [p:34,w:1,m:0] [uWSGIWorker1Core1] Loaded tool data table 'all_fasta' from file '/data/shed_tools/toolshed.g2.bx.psu.edu/repos/brinkmanlab/microbedb/40d14d5c8125/microbedb/tool_data_table_conf.xml.sample'
galaxy.tool_shed.metadata.metadata_generator DEBUG 2020-11-09 01:08:25,828 [p:34,w:1,m:0] [uWSGIWorker1Core1] Loaded Data Manager tool_files: all_fasta.xml
galaxy.util ERROR 2020-11-09 01:08:26,163 [p:34,w:1,m:0] [uWSGIWorker1Core2] ERROR: Unable to read builds file: [Errno 2] No such file or directory: 'tool-data/shared/ucsc/builds.txt.sample'
galaxy.tools.data DEBUG 2020-11-09 01:08:26,740 [p:34,w:1,m:0] [uWSGIWorker1Core1] Error attempting to parse file tool_data_table_conf.xml: expected str, bytes or os.PathLike object, not NoneType
Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/tools/data/__init__.py", line 158, in add_new_entries_from_config_file
    table_elems = self.load_from_config_file(config_filename=config_filename,
  File "/srv/galaxy/lib/galaxy/tools/data/__init__.py", line 123, in load_from_config_file
    table = ToolDataTable.from_elem(table_elem, tool_data_path, from_shed_config, filename=filename, tool_data_path_files=self.tool_data_path_files, other_config_dict=self.other_config_dict)
  File "/srv/galaxy/lib/galaxy/tools/data/__init__.py", line 248, in from_elem
    return tool_data_table_types[table_type](table_elem, tool_data_path, from_shed_config=from_shed_config, filename=filename, tool_data_path_files=tool_data_path_files, other_config_dict=other_config_dict)
  File "/srv/galaxy/lib/galaxy/tools/data/__init__.py", line 333, in __init__
    self.configure_and_load(config_element, tool_data_path, from_shed_config)
  File "/srv/galaxy/lib/galaxy/tools/data/__init__.py", line 401, in configure_and_load
    corrected_filename = os.path.join(self.tool_data_path, file_name)
  File "/usr/local/lib/python3.8/posixpath.py", line 76, in join
    a = os.fspath(a)
TypeError: expected str, bytes or os.PathLike object, not NoneType
galaxy.tools.data DEBUG 2020-11-09 01:08:26,740 [p:34,w:1,m:0] [uWSGIWorker1Core1] ToolDataTableManager.to_xml_file called without any elements to add or remove.
galaxy.tool_shed.galaxy_install.metadata.installed_repository_metadata_manager ERROR 2020-11-09 01:08:26,776 [p:34,w:1,m:0] [uWSGIWorker1Core1] Error while loading tool at path '/data/shed_tools/toolshed.g2.bx.psu.edu/repos/brinkmanlab/microbedb/40d14d5c8125/microbedb/all_fasta.xml'
Traceback (most recent call last):
  File "/srv/galaxy/lib/galaxy/tool_shed/galaxy_install/metadata/installed_repository_metadata_manager.py", line 77, in get_repository_tools_tups
    tool = self.app.toolbox.load_tool(os.path.abspath(load_relative_path), guid=guid, use_cached=False)
  File "/srv/galaxy/lib/galaxy/tools/toolbox/base.py", line 856, in load_tool
    tool = self.create_tool(config_file=config_file, tool_shed_repository=tool_shed_repository, guid=guid, tool_cache_data_dir=tool_cache_data_dir, **kwds)
  File "/srv/galaxy/lib/galaxy/tools/__init__.py", line 327, in create_tool
    cache.set(config_file, tool_source)
  File "/srv/galaxy/lib/galaxy/tools/cache.py", line 96, in set
    self._cache[config_file] = to_persist
  File "/srv/galaxy/venv/lib/python3.8/site-packages/sqlitedict.py", line 249, in __setitem__
    raise RuntimeError('Refusing to write to read-only SqliteDict')
RuntimeError: Refusing to write to read-only SqliteDict
galaxy.tool_shed.galaxy_install.tools.data_manager ERROR 2020-11-09 01:08:26,785 [p:34,w:1,m:0] [uWSGIWorker1Core1] Data manager tool guid 'toolshed.g2.bx.psu.edu/repos/brinkmanlab/microbedb/microbedb_all_fasta/1.0' could not be found for 'microbedb_all_fasta'. Perhaps the tool is invalid?
sqlitedict INFO 2020-11-09 01:08:26,813 [p:34,w:1,m:0] [uWSGIWorker1Core2] opening Sqlite table 'unnamed' in '/data/tool_cache/tmpgrcokzjvcache.sqlite.tmp'
sqlitedict.SqliteMultithread DEBUG 2020-11-09 01:08:26,882 [p:34,w:1,m:0] [Thread-46] received: --close--, send: --no more--
10.0.1.132 - - [09/Nov/2020:01:08:11 +0000] "POST /api/tool_shed_repositories/new/install_repository_revision HTTP/1.1" 200 - "-" "terraform-provider-galaxy/galaxy - blend4go (devel)"
```

--- a/lib/galaxy/tools/cache.py
+++ b/lib/galaxy/tools/cache.py
@@ -93,7 +93,10 @@ class ToolDocumentCache:
                     'paths_and_modtimes': tool_source.paths_and_modtimes(),
                     'tool_cache_version': CURRENT_TOOL_CACHE_VERSION,
                 }
-                self._cache[config_file] = to_persist
+                try:
+                    self._cache[config_file] = to_persist
+                except RuntimeError:
+                    log.debug("Tool document cache not writeable")
         except sqlite3.OperationalError:
             log.debug("Tool document cache unavailable")
 
@@ -102,7 +105,7 @@ class ToolDocumentCache:
             self._make_writable()
             try:
                 del self._cache[config_file]
-            except KeyError:
+            except (KeyError, RuntimeError):
                 pass
 
     def __del__(self):

